packages/shared/src/models/IPlaygroundMessage.ts

```diff
@@ -34,13 +34,13 @@ export interface ModelUsage {
 
 export interface ChatMessage extends Message {
   role: 'system' | 'user' | 'assistant';
-  content?: string;
-  usage?: ModelUsage;
+  content?: string | object;
 }
 
 export interface AssistantChat extends ChatMessage {
   role: 'assistant';
   completed?: number;
+  content?: string | ToolCall;
```

Comment:
Why not using a dedicated properties, like `tools` ? trying to put everything into content may lead to type issues, and force us to always check

Reply:
Because this is how the inference server returns the assistant's response.
I didn't want to diverge much from the upstream behaviors.

Reply:
If this is how the server respond I am good with it, forget my comments about it